PYME/Analysis/points/coordinate_tools.py

```diff
@@ -0,0 +1,282 @@
+        cartesian vector orthogonal to the axial axis
+    v2 : ndarray
+        cartesian vector othogonal to the previous two
+
+    Returns
+    -------
+
+    """
+    # make sure we have a reasonable task
+    assert ((length > 0) and (radius > 0))
+    # make sure all input vectors are unit vectors
+    v_0 = v0 / np.linalg.norm(v0)
+    v_1 = v1 / np.linalg.norm(v1)
+    v_2 = v2 / np.linalg.norm(v2)
+    # make vectors going from the base/center of the cylinder to all of the points we are checking
+    x_c, y_c, z_c = x - x0, y - y0, z - z0
+    positions_c = np.stack([x_c, y_c, z_c]).T
+    # check if they are within the ends of the cylinder, i.e. if the axial component is positive and less than length
+    axial_component = np.dot(positions_c, v_0)
+    inside_axially = np.logical_and(np.greater_equal(axial_component, 0.), np.less_equal(axial_component, length))
+    # now check if they're inside radially
+    radial_component0 = np.dot(positions_c[inside_axially], v_1)
+    radial_component1 = np.dot(positions_c[inside_axially], v_2)
+    # just in case we're working with a ton of points, throw out the ones which are obviously too far out
+    worth_checking = np.logical_and(np.less_equal(radial_component0, radius), np.less_equal(radial_component1, radius))
+    # now do the slow bit
+    inside_r = np.less_equal(np.sqrt(radial_component0[worth_checking] ** 2 + radial_component1[worth_checking] ** 2), radius)
+
+    # handle the indexing
+    inside = np.zeros_like(x, dtype=bool)
+    # can't slice because this copies, use where as a sort of disgusting work around
+    inside[np.where(inside_axially)[0][np.where(worth_checking)[0]][inside_r]] = True
+    return inside, axial_component[inside]
+
+
+def pixel_index_of_points_in_image(image, points):
+    """
+    Map positions into indices of an image
+
+    Parameters
+    ----------
+    image: PYME.IO.ImageStack
+        image with complete metadata
+    points: PYME.IO.tabular.TabularBase
+
+    Returns
+    -------
+    x_index: ndarray
+        x pixel index in image for each point
+    y_index: ndarray
+        y pixel index in image for each point
+    z_index: ndarray
+        z pixel index in image for each point
+
+    """
+    from PYME.IO.MetaDataHandler import get_camera_roi_origin
+
+    x0, y0, z0 = image.origin
+
+    # account for point data ROIs
+    try:
+        roi_x0, roi_y0 = get_camera_roi_origin(points.mdh)
+
+        p_ox = roi_x0 * points.mdh['voxelsize.x'] * 1e3
+        p_oy = roi_y0 * points.mdh['voxelsize.y'] * 1e3
+    except AttributeError:
+        raise RuntimeError('metadata specifying ROI position and voxelsize are missing')
+
+    # Image origin is referenced to top-left corner of pixelated image.
+    # FIXME - localisations are currently referenced to centre of raw pixels
+    x_index = np.floor((points['x'] + p_ox - x0) / image.pixelSize).astype('i')
+    y_index = np.floor((points['y'] + p_oy - y0) / image.pixelSize).astype('i')
```

Comment:
`image.pixelSize` is just the x axis size (which is usually OK, but not 100% safe in case you have different pixel sizes in x and y. Probably better to use `image.voxelsize_nm.y` etc ...

Reply:
good catch, now taken care of